### PR TITLE
chore:  Reorganize develop docs and change sidebar ordering

### PR DESCRIPTION
**Pull request by markstory**
## DESCRIBE YOUR PR

The develop docs organization has grown organically and could benefit from some more cohesive organization. The new top level directories roughly map out to:

- getting started : for general information on philosophy and inclusive language
- development : getting sentry working locally, contributing code an working with local development services.
- application : application design and architecture documentation.
- frontend : frontend documentation - unchanged
- backend : I've moved a bunch of documentation here. Previously 'services' was used for *both* application services and standalone applications (services) that compose sentry. Now backend contains everything you'd find in the python monolith, while 'services' contains standalone applications.
- services : services we operate that exist outside of the monolith
- integrations : unchanged
- sdk : unchanged
- relay : unchanged

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

### CODE REVIEW

File: develop-docs/application/config.mdx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Configuration
+sidebar_order: 60
```
Comment (markstory) [inline develop-docs/application/config.mdx:3]: I spaced most of the pages out by increments of 10 as it was easy to count and left holes for future docs to be added.

File: develop-docs/backend/transaction-clustering/index.mdx [hunk 2]
```diff
@@ -6,8 +6,9 @@ sidebar_order: 60
 Sentry attempts to scrub high-cardinality identifiers from URL transactions
 to aggregate performance data and provide more valuable insights.
 
-In terms of user experience, this feature plays a similar role as [Issue Grouping](/grouping/).
-In terms of technical implementation, it is similar to [Data Scrubbing](/pii/).
+In terms of user experience, this feature plays a similar role as [Issue Grouping](/backend/grouping/).
+In terms of technical implementation, it is similar to [Data
```
Comment (stephanie-anderson) [inline develop-docs/backend/transaction-clustering/index.mdx:10]: Is this new line intentional?

**Review by stephanie-anderson — COMMENTED** [review]
Big THANKS for investing some time in improving our develop docs! I noticed that you haven't set up any redirects, could you add them for the pages you've moved?

You can add those to this file:
https://github.com/getsentry/sentry-docs/blob/cbd9baaa4b3e7cb175a13c7895947bb1ed5306a9/redirects.js#L4

File: develop-docs/backend/transaction-clustering/index.mdx (hunk 2 above)
Comment (markstory) [reply develop-docs/backend/transaction-clustering/index.mdx:10]: Nope, that was a mistake.

**Comment by markstory** [comment]
> I noticed that you haven't set up any redirects, could you add them for the pages you've moved?

Updated! I didn't know there were redirects 😊

**Review by stephanie-anderson — APPROVED** [review]
looks great! thanks for adding the redirects 🙏

**Pull request merged**